cairo_programs/memory_integration_tests.cairo

```diff
@@ -0,0 +1,108 @@
+from starkware.cairo.common.alloc import alloc
+from starkware.cairo.common.bool import TRUE, FALSE
+from starkware.cairo.common.memcpy import memcpy
+from starkware.cairo.common.memset import memset 
+
+# Helper functions
+func fill_array(array : felt*, base : felt, step : felt, array_length : felt, iterator : felt):
+    if iterator == array_length:
+        return()
+    end
+    assert array[iterator] = base + step * iterator
+    return fill_array(array, base, step, array_length, iterator + 1)
+end
+
+func check_array(array : felt*, value : felt, array_length : felt, iterator : felt ) -> (r : felt):
+    if iterator == array_length:
+        return(TRUE)
+    end
+    if array[iterator] != value:
+        return(FALSE)
+    end
+    return check_array(array, value, array_length, iterator + 1)
+end
+
+func compare_arrays(array_a : felt*, array_b : felt*, array_length : felt, iterator : felt ) -> (r : felt):
+    if iterator == array_length:
+        return(TRUE)
+    end
+    if array_a[iterator] != array_b[iterator]:
+        return(FALSE)
+    end
+    return compare_arrays(array_a, array_b, array_length, iterator + 1)
+end
+
+# ---------------------------------------------------------------------------------- #
+
+# Test functions
+func test_memcpy(src : felt*, len : felt, iter : felt) -> ():
+    alloc_locals
+    if iter == len:
+        return ()
+    end
+
+    let (dst : felt*) = alloc()
+    memcpy(dst=dst, src=src, len=len)
+
+    let result : felt = compare_arrays(src, dst, len, 0)
+    assert result = TRUE
+
+    return test_memcpy(src, len, iter + 1)
```

Comment:
I don't see the point in iterating over the same function with the same inputs.
I think it is better to run the function once, and if you want to modify the function execution time you pass a greater `len` as an input.